phpBB/includes/functions_admin.php

```diff
@@ -2573,6 +2573,11 @@ function prune($forum_id, $prune_mode, $prune_date, $prune_flags = 0, $auto_sync
 		$sql_and .= ' AND topic_status = ' . ITEM_MOVED . " AND topic_last_post_time < $prune_date";
 	}
 
+	if ($prune_limit > 0)
+	{
+		$sql_and .= " LIMIT $prune_limit";
```

Comment:
What I'm talking about is the current implementation of sql_query_limit().

In `phpBB\phpbb\db\driver\mysql_base.php`
```
	/**
	* Build LIMIT query
	*/
	function _sql_query_limit($query, $total, $offset = 0, $cache_ttl = 0)
	{
		$this->query_result = false;

		// if $total is set to 0 we do not want to limit the number of rows
		if ($total == 0)
		{
			// MySQL 4.1+ no longer supports -1 in limit queries
			$total = '18446744073709551615';
		}

		$query .= "\n LIMIT " . ((!empty($offset)) ? $offset . ', ' . $total : $total);

		return $this->sql_query($query, $cache_ttl);
	}
```

I do not think this is an ideal implementation.  But I don't know if they had a valid reason for always adding the LIMIT to the query.  Because in my implementation of the limit, $prune_limit can be 0 to specify that it should attempt to prune everything in one transaction.

If you still feel I should use sql_query_limit(), I shall.  Because I understand where you are coming from for different DBs.  And it maybe better to address the unneeded addition of LIMIT in sql_query_limit() instead.